zstacklib/zstacklib/gpu/vendors/nvidia.py

```
@@ -312,6 +312,103 @@ def _parse_vgpu_output(output):
+                result[pid] = (sm, fb_mib)
+            except (ValueError, IndexError):
+                continue
+        return result
+
+    @staticmethod
+    def _get_worker_host_pid(worker):
+        """Get the host-visible PID for a TF worker (handles container mode)."""
+        if worker.container_id:
+            import subprocess
+            try:
+                o = subprocess.check_output(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)